src/DynamoCore/Search/SearchDictionary.cs

```diff
@@ -323,11 +323,33 @@ private static string[] SplitOnWhiteSpace(string s)
             return s.Split(null);
         }
 
-        private static bool ContainsSpecialCharacters(string element)
+        public static Char[] SpecialAndInvalidCharacters()
         {
-            return element.Contains("*") || element.Contains(".") || element.Contains(" ")
-                || element.Contains("\\");
+            char[] invalidFileChars = System.IO.Path.GetInvalidFileNameChars();
+
+            List<string> filteredChars = new List<string>();
+            foreach (char someChar in invalidFileChars)
+            {   // Excluding white spaces and uncommon characters, only keeping the displayed in the Windows alert
+                if (!Char.IsWhiteSpace(someChar) && (int)someChar > 31)
+                {
+                    filteredChars.Add(someChar.ToString());
+                }
+            }
+
+            char[] result = new char[filteredChars.Count];
+            for (int i = 0; i < filteredChars.Count; i++)
+            {
+                result[i] = char.Parse(filteredChars[i]);
+            }
+            return result;
```

Comment:
Two suggestions I have:
- An easier way may I suggest is to convert the char[] to string and replace the space with string.null and then convert it back to char[].
-  Also currently this SpecialAndInvalidCharacters will be calculated runtime everytime we publish a package, can we initialize it as a static property only once and get back to it everytime we are doing package name check?

Reply:
The SpecialAndInvalidCharacters function was reduced to a single line